src/node.ts

```diff
@@ -0,0 +1,184 @@
+class ReferencedValue {
```

Comment:
`ReferencedValue` and `MemoryPointer` can be type definitions instead which is more idiomatic for typescript:

```
type ReferencedValue = {
  dataPointer: MemoryPointer;
  value: Buffer;
};

type MemoryPointer = {
  offset: number;
  length: number;
};
```